commitizen/commands/init.py

```diff
@@ -19,34 +21,43 @@ def __init__(self, config: BaseConfig, *args):
         self.cz = factory.commiter_factory(self.config)
 
     def __call__(self):
-        values_to_add = {}
+        if self.config.path:
+            out.line(f"Config file {self.config.path} already exists")
+            return
 
         # No config for commitizen exist
-        if not self.config.path:
-            config_path = self._ask_config_path()
-            if "toml" in config_path:
-                self.config = TomlConfig(data="", path=config_path)
-            elif "json" in config_path:
-                self.config = JsonConfig(data="{}", path=config_path)
-            elif "yaml" in config_path:
-                self.config = YAMLConfig(data="", path=config_path)
-
-            self.config.init_empty_config_content()
-
-            values_to_add["name"] = self._ask_name()
-            tag = self._ask_tag()
-            values_to_add["version"] = Version(tag).public
-            values_to_add["tag_format"] = self._ask_tag_format(tag)
-            self._update_config_file(values_to_add)
-
-            if questionary.confirm("Do you want to install pre-commit hook?").ask():
-                self._install_pre_commit_hook()
-
-            out.write("You can bump the version and create changelog running:\n")
-            out.info("cz bump --changelog")
-            out.success("The configuration are all set.")
-        else:
-            out.line(f"Config file {self.config.path} already exists")
+        config_path = self._ask_config_path()
+        if "toml" in config_path:
+            self.config = TomlConfig(data="", path=config_path)
+        elif "json" in config_path:
+            self.config = JsonConfig(data="{}", path=config_path)
+        elif "yaml" in config_path:
+            self.config = YAMLConfig(data="", path=config_path)
+        self.config.init_empty_config_content()
+
+        values_to_add = {}
+        values_to_add["name"] = self._ask_name()
+        tag = self._ask_tag()
+        values_to_add["version"] = Version(tag).public
+        values_to_add["tag_format"] = self._ask_tag_format(tag)
+        self._update_config_file(values_to_add)
+
+        hook_types = questionary.checkbox(
+            "What types of pre-commit hook you want to install? (Leave blank if you don't want to install)",
+            choices=[
+                questionary.Choice("commit-msg", checked=True),
+                questionary.Choice("pre-push", checked=True),
+            ],
+        ).ask()
+        if hook_types:
+            if not self._install_pre_commit_hook(hook_types):
```

Comment:
I think we should put the following code into `_exec_install_pre_commit_hook`. This seems to be the responsibility of that function. @Bogay what do you think?

```python
                raise InitFailedError(
                    "Installation failed. See error outputs for more information."
                )
```